File: backend/src/main/java/com/ameri/objects/beans/editorBeans/EditorBeans.java
package com.ameri.objects.beans.editorBeans;

import com.ameri.objects.classes.user.editor.Comment;
import com.ameri.objects.classes.user.editor.EditorAccount;
import com.ameri.objects.classes.user.editor.Subscription;

import java.util.List;

public class EditorBeans {

    private double total;
    private List<MagazineBeans> magazineBeansList;
    private List<Subscription> subscriptionList;
    private List<Comment> commentList;
    private List<EditorAccount> editorAccountList;

    public EditorBeans(){}

    public double getTotal() {
        return total;
    }

    public void setTotal(double total) {
        this.total += total;
    }

    public List<MagazineBeans> getMagazineBeansList() {
        return magazineBeansList;
    }

    public void setMagazineBeansList(List<MagazineBeans> magazineBeansList) {
        this.magazineBeansList = magazineBeansList;
    }

    public List<Subscription> getSubscriptionList() {
        return subscriptionList;
    }

    public void setSubscriptionList(List<Subscription> subscriptionList) {
        this.subscriptionList = subscriptionList;
    }

    public List<Comment> getCommentList() {
        return commentList;
    }

    public void setCommentList(List<Comment> commentList) {
        this.commentList = commentList;
    }

    public List<EditorAccount> getEditorAccountList() {
        return editorAccountList;
    }

    public void setEditorAccountList(List<EditorAccount> editorAccountList) {
        this.editorAccountList = editorAccountList;
    }
}

File: frontend/src/objects/classes/beans/editorBeans/EditorBeans.ts
import {MagazineBeans} from "./MagazineBeans";
import {Subscription} from "../../usuario/editor/Subscription";
import {Comment} from "../../usuario/editor/Comment";
import {EditorAccount} from "../../usuario/editor/EditorAccount";

export class EditorBeans{

  total: number;
  magazineBeansList: Array<MagazineBeans> ;
  subscriptionList: Array<Subscription>;
  commentList: Array<Comment>;
  editorAccountList: Array<EditorAccount>;


  constructor(total: number, magazineBeansList: Array<MagazineBeans>, subscriptionList: Array<Subscription>, commentList: Array<Comment>, editorAccountList: Array<EditorAccount>) {
    this.total = total;
    this.magazineBeansList = magazineBeansList;
    this.subscriptionList = subscriptionList;
    this.commentList = commentList;
    this.editorAccountList = editorAccountList;
  }
}
